src/types/BaseField.ts

```diff
@@ -56,9 +60,68 @@ const writableBaseFieldSchema: JSONSchemaType<WritableBaseField> = {
 
 const isWritableBaseField = ajv.compile(writableBaseFieldSchema);
 
+type WritableBaseFieldWithLocalizationsContext = Writable<BaseField> & {
+	localizations: Record<string, InternallyWritableBaseFieldLocalization>;
+};
+
+const writableBaseFieldWithLocalizationsContextSchema: JSONSchemaType<WritableBaseFieldWithLocalizationsContext> =
+	{
+		type: 'object',
+		properties: {
+			label: {
+				type: 'string',
+			},
+			description: {
+				type: 'string',
+			},
+			shortCode: {
+				type: 'string',
+			},
+			dataType: {
+				type: 'string',
+				enum: Object.values(BaseFieldDataType),
+			},
+			scope: {
+				type: 'string',
+				enum: Object.values(BaseFieldScope),
+			},
+			localizations: {
+				type: 'object',
+				additionalProperties: internallyWritableBaseFieldLocalizationSchema,
+				required: [],
+			},
+		},
+		required: [
+			'label',
+			'description',
+			'shortCode',
+			'dataType',
+			'scope',
+			'localizations',
+		],
+		additionalProperties: true,
+	};
+
+type WritableBaseFieldWithLocalizationsContextArray =
```

Comment:
I mention this later, but just to be clear: you don't want this type, nor do you want the associated schema and guard.